### ISSUE TITLE
Select with fromQuery & callable: unable to preselect value

### ISSUE THREAD

**Issue by syphernl** [formers/former#495]
In our select we required a `data-icon` field so we had to switch our normal select:

``` php
{!! Former::select('project_id')->label('Project')->required()->data_placeholder('Select a project')->placeholder('Select a project')->fromQuery(App\Models\Project::getAllVisible(),'title','id', isset($credential) ? $credential->project_id : null) !!}
```

to one with a callable instead:

``` php
{!! Former::select('project_id')->label('Project')->required()->data_placeholder('Select a project')->placeholder('Select a project')->fromQuery(App\Models\Project::getAllVisible(),function($model) {return $model->title;},['value' => 'id','data-icon' => function($model) {return $model->getIcon();}], isset($credential) ? $credential->project_id : null) !!}
```

However, after doing this it we're unable to preselect a value either via the fourth argument or via `select()` or `selectValue()`

**Comment by claar** [comment]
Wow, I had no idea callables were even supported. You're on your own figuring this out I'm afraid unless someone else wants to take a stab at this. PRs welcome!

**Comment by syphernl** [comment]
Perhaps @NelinD can be of help here?

**Comment by syphernl** [comment]
@claar It looks like the changes in PR #472 resolves this problem! It would be nice if this one could be merged in.

**Comment by claar** [comment]
@syphernl That PR is just waiting on tests to show what it fixes; feel free to copy his code to a new PR and add tests if you want to move it along!

**Comment by claar** [comment]
I can provide workarounds for this that don't use callables if this is still an issue -- please comment and let me know.
